{"relationshipsSections":[{"title":"Conforms To","kind":"relationships","identifiers":["doc:\/\/com.vigram.VigramSDK\/10Foundation14LocalizedErrorP","doc:\/\/com.vigram.VigramSDK\/s8CopyableP","doc:\/\/com.vigram.VigramSDK\/s5ErrorP","doc:\/\/com.vigram.VigramSDK\/s8SendableP"],"type":"conformsTo"}],"identifier":{"url":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError","interfaceLanguage":"swift"},"metadata":{"navigatorTitle":[{"text":"AuthenticationError","kind":"identifier"}],"role":"symbol","modules":[{"name":"VigramSDK"}],"roleHeading":"Enumeration","title":"AuthenticationError","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AuthenticationError","kind":"identifier"}],"externalID":"s:9VigramSDK19AuthenticationErrorO","symbolKind":"enum"},"kind":"symbol","abstract":[{"text":"AuthenticationError contains custom failure events when connecting to Authentication process.","type":"text"}],"sections":[],"seeAlsoSections":[{"title":"Errors","anchor":"Errors","identifiers":["doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripServiceError","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTaskError","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/PeripheralError","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/SinglePointRecordingError"],"generated":true}],"hierarchy":{"paths":[["doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK"]]},"schemaVersion":{"patch":0,"major":0,"minor":3},"topicSections":[{"title":"Enumeration Cases","anchor":"Enumeration-Cases","identifiers":["doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/bundleNotExist(_:)","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/failure(_:)","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/tokenNotExist(_:)","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/wrongCombination(_:)","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/wrongTokenAndBundle(_:)"],"generated":true},{"title":"Default Implementations","anchor":"Default-Implementations","identifiers":["doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/Error-Implementations","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/LocalizedError-Implementations"],"generated":true}],"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AuthenticationError","kind":"identifier"}],"platforms":["iOS"],"languages":["swift"]}]},{"content":[{"anchor":"overview","type":"heading","level":2,"text":"Overview"},{"type":"aside","content":[{"inlineContent":[{"text":"It is not ensured that calls to ","type":"text"},{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Authentication","isActive":true,"type":"reference"},{"text":" can only result in ","type":"text"},{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError","isActive":true,"type":"reference"},{"text":" events.","type":"text"}],"type":"paragraph"}],"name":"Note","style":"note"}],"kind":"content"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/vigramsdk\/authenticationerror"]}],"references":{"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError/failure(_:)":{"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"failure","kind":"identifier"},{"text":"(","kind":"text"},{"text":"String","kind":"typeIdentifier","preciseIdentifier":"s:SS"},{"text":")","kind":"text"}],"kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/failure(_:)","url":"\/documentation\/vigramsdk\/authenticationerror\/failure(_:)","role":"symbol","title":"AuthenticationError.failure(_:)","type":"topic","abstract":[{"text":"Authentication is failure - Other error received from server","type":"text"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripTask":{"fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"NtripTask","kind":"identifier"}],"type":"topic","kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTask","url":"\/documentation\/vigramsdk\/ntriptask","abstract":[{"type":"text","text":"A NtripTask requests RTK data over a NTRIP connection."}],"navigatorTitle":[{"text":"NtripTask","kind":"identifier"}],"title":"NtripTask","role":"symbol"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError":{"kind":"symbol","role":"symbol","abstract":[{"type":"text","text":"AuthenticationError contains custom failure events when connecting to Authentication process."}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError","type":"topic","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AuthenticationError","kind":"identifier"}],"title":"AuthenticationError","navigatorTitle":[{"text":"AuthenticationError","kind":"identifier"}],"url":"\/documentation\/vigramsdk\/authenticationerror"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/BluetoothServiceError":{"navigatorTitle":[{"text":"BluetoothServiceError","kind":"identifier"}],"url":"\/documentation\/vigramsdk\/bluetoothserviceerror","abstract":[{"text":"BluetoothServiceError lists different errors that can be propagated by ","type":"text"},{"isActive":true,"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothService","type":"reference"},{"text":" implementations.","type":"text"}],"title":"BluetoothServiceError","kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"BluetoothServiceError","kind":"identifier"}],"type":"topic","role":"symbol"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError/tokenNotExist(_:)":{"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"tokenNotExist","kind":"identifier"},{"text":"(","kind":"text"},{"text":"String","kind":"typeIdentifier","preciseIdentifier":"s:SS"},{"text":")","kind":"text"}],"kind":"symbol","role":"symbol","url":"\/documentation\/vigramsdk\/authenticationerror\/tokennotexist(_:)","abstract":[{"type":"text","text":"Error authentication - token is not exist"}],"type":"topic","title":"AuthenticationError.tokenNotExist(_:)","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/tokenNotExist(_:)"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/PeripheralError":{"fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"PeripheralError"}],"abstract":[{"type":"text","text":"PeripheralError combines different failure events that can be emitted on sending data to a "},{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Peripheral","isActive":true},{"type":"text","text":"."}],"type":"topic","title":"PeripheralError","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/PeripheralError","navigatorTitle":[{"kind":"identifier","text":"PeripheralError"}],"role":"symbol","url":"\/documentation\/vigramsdk\/peripheralerror","kind":"symbol"},"logo.png":{"variants":[{"url":"\/images\/com.vigram.VigramSDK\/logo.png","traits":["1x","light"]}],"identifier":"logo.png","alt":"Vigram logo.","type":"image"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError/bundleNotExist(_:)":{"role":"symbol","url":"\/documentation\/vigramsdk\/authenticationerror\/bundlenotexist(_:)","kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/bundleNotExist(_:)","type":"topic","fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"bundleNotExist","kind":"identifier"},{"text":"(","kind":"text"},{"text":"String","preciseIdentifier":"s:SS","kind":"typeIdentifier"},{"text":")","kind":"text"}],"abstract":[{"text":"Error authentication - bundle is not exist","type":"text"}],"title":"AuthenticationError.bundleNotExist(_:)"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripTaskError":{"type":"topic","url":"\/documentation\/vigramsdk\/ntriptaskerror","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NtripTaskError"}],"abstract":[{"type":"text","text":"NtripTaskError lists different errors that can be emitted by "},{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTask","isActive":true},{"type":"text","text":" implementations."}],"kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTaskError","title":"NtripTaskError","navigatorTitle":[{"kind":"identifier","text":"NtripTaskError"}],"role":"symbol"},"doc://com.vigram.VigramSDK/10Foundation14LocalizedErrorP":{"title":"Foundation.LocalizedError","type":"unresolvable","identifier":"doc:\/\/com.vigram.VigramSDK\/10Foundation14LocalizedErrorP"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripServiceError":{"url":"\/documentation\/vigramsdk\/ntripserviceerror","title":"NtripServiceError","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripServiceError","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NtripServiceError"}],"type":"topic","kind":"symbol","abstract":[{"text":"NtripServiceError contains custom failure events when connecting to Ntrip services.","type":"text"}],"navigatorTitle":[{"kind":"identifier","text":"NtripServiceError"}],"role":"symbol"},"doc://com.vigram.VigramSDK/s5ErrorP":{"title":"Swift.Error","identifier":"doc:\/\/com.vigram.VigramSDK\/s5ErrorP","type":"unresolvable"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/Peripheral":{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Peripheral","abstract":[{"type":"text","text":"Peripheral provides a write and receive interface to communicate with the viDoc device."}],"kind":"symbol","type":"topic","role":"symbol","url":"\/documentation\/vigramsdk\/peripheral","title":"Peripheral","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"Peripheral"}],"navigatorTitle":[{"kind":"identifier","text":"Peripheral"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK":{"kind":"symbol","role":"collection","type":"topic","url":"\/documentation\/vigramsdk","title":"VigramSDK","abstract":[],"images":[{"type":"icon","identifier":"logo.png"}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError/Error-Implementations":{"abstract":[],"kind":"article","url":"\/documentation\/vigramsdk\/authenticationerror\/error-implementations","title":"Error Implementations","type":"topic","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/Error-Implementations","role":"collectionGroup"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/SinglePointRecordingError":{"kind":"symbol","type":"topic","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/SinglePointRecordingError","title":"SinglePointRecordingError","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"SinglePointRecordingError"}],"navigatorTitle":[{"kind":"identifier","text":"SinglePointRecordingError"}],"role":"symbol","url":"\/documentation\/vigramsdk\/singlepointrecordingerror","abstract":[{"text":"SinglePointRecordingError are thrown, when due to the filtering phase of single point recordings there is not enough valid data to create a singlePoint.","type":"text"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError/LocalizedError-Implementations":{"role":"collectionGroup","abstract":[],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/LocalizedError-Implementations","type":"topic","url":"\/documentation\/vigramsdk\/authenticationerror\/localizederror-implementations","title":"LocalizedError Implementations","kind":"article"},"doc://com.vigram.VigramSDK/s8SendableP":{"title":"Swift.Sendable","type":"unresolvable","identifier":"doc:\/\/com.vigram.VigramSDK\/s8SendableP"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError/wrongCombination(_:)":{"abstract":[{"type":"text","text":"Error authentication - wrong combination (bundle, token or platformId is incorrect)"}],"role":"symbol","url":"\/documentation\/vigramsdk\/authenticationerror\/wrongcombination(_:)","title":"AuthenticationError.wrongCombination(_:)","type":"topic","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"wrongCombination"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"kind":"text","text":")"}],"kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/wrongCombination(_:)"},"doc://com.vigram.VigramSDK/s8CopyableP":{"type":"unresolvable","identifier":"doc:\/\/com.vigram.VigramSDK\/s8CopyableP","title":"Swift.Copyable"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/Authentication":{"url":"\/documentation\/vigramsdk\/authentication","abstract":[{"type":"text","text":"First creates a "},{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Authentication","isActive":true},{"type":"text","text":" to check token and use SDK functions."}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Authentication","title":"Authentication","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"Authentication"}],"kind":"symbol","role":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"Authentication"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK/BluetoothService":{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothService","abstract":[{"text":"A BluetoothService is used to handle bluetooth connections.","type":"text"}],"kind":"symbol","type":"topic","role":"symbol","url":"\/documentation\/vigramsdk\/bluetoothservice","title":"BluetoothService","fragments":[{"kind":"keyword","text":"protocol"},{"text":" ","kind":"text"},{"text":"BluetoothService","kind":"identifier"}],"navigatorTitle":[{"text":"BluetoothService","kind":"identifier"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError/wrongTokenAndBundle(_:)":{"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"wrongTokenAndBundle","kind":"identifier"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":")","kind":"text"}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError\/wrongTokenAndBundle(_:)","kind":"symbol","abstract":[{"type":"text","text":"Error authentication - wrong combination (bundle and token is incorrect)"}],"url":"\/documentation\/vigramsdk\/authenticationerror\/wrongtokenandbundle(_:)","role":"symbol","type":"topic","title":"AuthenticationError.wrongTokenAndBundle(_:)"}}}